src/System.Management.Automation/engine/parser/Compiler.cs

```diff
@@ -6492,17 +6537,25 @@ public Expression SetValue(Compiler compiler, Expression rhs)
             if (memberNameAst != null)
             {
                 string name = memberNameAst.Value;
-                return DynamicExpression.Dynamic(
+                var expr = DynamicExpression.Dynamic(
                                             PSSetMemberBinder.Get(name, compiler._memberFunctionType, MemberExpression.Static),
                                             typeof(object),
                                             CachedTarget ?? GetTargetExpr(compiler), rhs);
+
+                return MemberExpression.NullConditionalAccess ?
+                    Compiler.GetNullCheckExpressionBlock(CachedTarget ?? GetTargetExpr(compiler), expr)
```

Comment:
Are you allowing assignment to member expression with null-conditional operator? This should be banned at the semantic checking step.

Reply:
Removed.